onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -376,22 +377,20 @@ static Status RegisterTensorrtKernels(KernelRegistry& kernel_registry) {
   return Status::OK();
 }
 
-static std::shared_ptr<KernelRegistry> s_kernel_registry;
+static std::shared_ptr<KernelRegistry> CreateTensorrtKernelRegistry() {
```

Comment:
It could throw. But you won't want to see any exception was thrown before the main function started.  That's why in CPU EP's implementation, it also returns a onnxruntime::Status.

Reply:
The CPU EP's GetKernelRegistry() may also throw.
https://github.com/microsoft/onnxruntime/blob/7b5464ed7bdd4eda5ea33977c008538b08da9025/onnxruntime/core/providers/cpu/cpu_execution_provider.cc#L2059

The CreateXKernelRegistry() functions added here are static to the .cc files and only called by the EP's GetKernelRegistry().

If GetKernelRegistry() is called prior to main, then there could be a problem with the exception.